framework/src/main/java/org/checkerframework/framework/util/AnnotatedTypes.java

```diff
@@ -968,37 +968,31 @@ public static List<AnnotatedTypeMirror> adaptParameters(
             List<? extends ExpressionTree> args) {
         List<AnnotatedTypeMirror> parameters = method.getParameterTypes();
 
-        if (parameters.isEmpty()) {
-            return parameters;
-        }
-
         // Handle anonymous constructors that extend a class with an enclosing type.
         if (method.getElement().getKind() == ElementKind.CONSTRUCTOR
                 && method.getElement().getEnclosingElement().getSimpleName().contentEquals("")) {
             DeclaredType t =
                     TypesUtils.getSuperClassOrInterface(
                             method.getElement().getEnclosingElement().asType(), atypeFactory.types);
-            if (t.getEnclosingType() != null) {
-                if (args.isEmpty()) {
-                    // TODO: ugly hack to attempt to fix mismatch
-                    parameters = parameters.subList(1, parameters.size());
-                } else {
-                    TypeMirror p0tm = parameters.get(0).getUnderlyingType();
-                    // Is the first parameter either equal to the enclosing type?
-                    if (atypeFactory.types.isSameType(t.getEnclosingType(), p0tm)) {
-                        // Is the first argument the same type as the first parameter?
-                        if (!atypeFactory.types.isSameType(TreeUtils.typeOf(args.get(0)), p0tm)) {
-                            // Remove the first parameter.
-                            parameters = parameters.subList(1, parameters.size());
-                        }
-                    }
-                }
-                if (parameters.isEmpty()) {
-                    return parameters;
-                }
+            // Let parameterTypes and arguments match, i.e., size, and we only handle Java versions
+            // below 11 since they have an extra enclosing expression argument
+            if (t.getEnclosingType() != null
+                    && SystemUtil.jreVersion < 11
+                    && args.size() > 0
+                    && atypeFactory.types.isSameType(
+                            t.getEnclosingType(),
+                            atypeFactory.getAnnotatedType(args.get(0)).getUnderlyingType())) {
+                List<AnnotatedTypeMirror> p = new ArrayList<>(parameters.size() + 1);
+                p.add(atypeFactory.getAnnotatedType(args.get(0)));
+                p.addAll(1, parameters);
```

Comment:
Is the `1` necessary? Plain `addAll` appends to the end and at this point `p` contains only one element anyways, right?

Reply:
Thanks.No, we don't need it.